src/sortStudents.ts

```diff
@@ -1,16 +1,58 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  return [...students].sort((a: Student, b: Student): number => {
+    const averageGradeA: number = a.grades.reduce(
+      (acc: number, grade: number) => acc + grade,
+    ) / a.grades.length;
+    const averageGradeB: number = b.grades.reduce(
+      (acc: number, grade: number) => acc + grade,
+    ) / b.grades.length;
+    const marriedA = a.married ? 1 : 0;
+    const marriedB = b.married ? 1 : 0;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    switch (sortBy) {
+      case SortType.Name:
+        return order === 'asc'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name);
+      case SortType.Surname:
```

Comment:
add spacing between blocks
```suggestion
          : b.name.localeCompare(a.name);

      case SortType.Surname:
```